runtime/src/main/java/io/quarkiverse/ironjacamar/runtime/endpoint/DefaultMessageEndpointFactory.java

```diff
@@ -65,11 +70,12 @@ public MessageEndpoint createEndpoint(XAResource xaResource, long timeout) throw
 
     @Override
     public MessageEndpoint createEndpoint(XAResource xaResource) {
+        Context rootContext = vertx.getOrCreateContext();
         MessageEndpoint endpoint;
         if (xaResource == null) {
             endpoint = NoopMessageEndpoint.INSTANCE;
         } else {
-            endpoint = new TransactionAwareMessageEndpoint(xaResource);
+            endpoint = new TransactionAwareMessageEndpoint(rootContext, xaResource);
```

Comment:
For maintenance purposes I think it's better if we had a `DuplicateContextMessageEndpoint` and wrap the existing endpoint before returning